### ISSUE TITLE
[BUG][Fuzzer][wrong-code] tl.ThreadSync silently drops the WAW __syncthreads() between a T.copy-in and a nonzero-offset T.fill of the same shared buffer → fill silently dropped

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2698]
A `T.fill`/`T.clear` of a **nonzero-offset sub-region** of a shared buffer that was just written by a `T.copy`-in silently produces wrong results: the fill is dropped and the consumer reads the copied-in values instead. The compiler accepts, lowers, and runs the program with no diagnostic, but omits the cross-thread `__syncthreads()` that the auto-barrier pass (`tl.ThreadSync`) is supposed to insert between the two conflicting writes. The identical pattern at offset 0 is handled correctly, so the compiler treats the idiom as legal and just mishandles the nonzero offset.

### Problem description

`tl.ThreadSync` fails to insert the required cross-thread write-after-write `__syncthreads()` between a `T.copy` into a shared buffer and a subsequent `T.fill` of a nonzero-offset sub-region of that buffer, so the fill is silently dropped and the buffer is consumed with stale copied-in values.

### Reproducible example

Minimal, self-contained. The trigger and the offset=0 control (which is correct) are in one script.

```python
import os
os.environ["TILELANG_CACHE_DIR"] = "/tmp/waw_repro/cache"

import torch
import tilelang
import tilelang.language as T

tilelang.disable_cache()

N = 64
OFF_LO, OFF_HI = 10, 16  # nonzero start offset -> triggers the bug

def build(off_lo, off_hi):
    @T.prim_func
    def main(A: T.Tensor((N,), "float32"), B: T.Tensor((N,), "float32")):
        with T.Kernel(1, threads=N) as _:
            As = T.alloc_shared((N,), "float32", scope="shared")
            T.copy(A, As)                       # copy-in : thread t writes As[t]
            T.fill(As[off_lo:off_hi], 99.0)     # fill    : threads write As[off_lo..]
            T.copy(As, B)                       # copy-out
    return tilelang.compile(main, out_idx=[1])

A = torch.arange(N, dtype=torch.float32, device="cuda")   # A = 0,1,2,...,63

# --- TRIGGER: nonzero offset [10:16] ---
B = build(OFF_LO, OFF_HI)(A)
print("offset=10 region  :", B[OFF_LO:OFF_HI].tolist())
# expected [99,99,99,99,99,99] ; got [10,11,12,13,14,15]  == the copied-in values

# --- CONTROL (a): offset 0 [0:6], same source shape ---
Bc = build(0, 6)(A)
print("offset=0  region  :", Bc[0:6].tolist())
# [99,99,99,99,99,99]  CORRECT
```

Output (deterministic, identical across 3 runs; tilelang 0.1.11, L40S sm_89):

```
offset=10 region  : [10.0, 11.0, 12.0, 13.0, 14.0, 15.0]   <- FILL SILENTLY DROPPED
offset=0  region  : [99.0, 99.0, 99.0, 99.0, 99.0, 99.0]   <- correct
```

### Traceback

No traceback — compiles and runs, silently wrong.

### Expected behavior

`B[10:16] == [99, 99, 99, 99, 99, 99]`. The offset=0 sibling `T.fill(As[0:6], 99.0)`, built from the **same source shape and the same copy-in**, produces `[99, 99, 99, 99, 99, 99]` correctly — so the compiler treats copy-in-then-partial-fill as a legal, supported idiom and merely mishandles the nonzero start offset. The generated CUDA for the offset=0 control emits a `__syncthreads()` *before* the fill; the offset=10 case drops exactly that barrier.

### Root cause

**`tl.ThreadSync` (`TileLangThreadSyncPlanner::FindConflict`) models the copy-in and the fill as writes from a *single shared thread variable*, which structurally cannot express a cross-thread WAW hazard; a nonzero fill offset then makes the two same-thread byte addresses provably disjoint, so the barrier is dropped.**

The generated CUDA is the smoking gun. Trigger (offset 10):

```cuda
As[threadIdx.x] = A[threadIdx.x];                     // copy-in: thread t writes As[t]
if (threadIdx.x < 6) { As[threadIdx.x + 10] = 99.0f; } // fill: threads 0..5 write As[10..15]
__syncthreads();                                       // the ONLY barrier — guards the copy-out read
B[threadIdx.x] = As[threadIdx.x];                      // copy-out
```

There is **no `__syncthreads()` between the copy-in and the fill.** Thread 10's copy-in writes `As[10]`; thread 0's fill writes `As[0+10] = As[10]` — two different threads, same address, unordered. The observed ordering (copy-in store winning over the fill store) is undefined; it merely happens to be stable on this GPU.

Control (a), offset 0, gets **two** barriers — one is inserted *before* the fill:

```cuda
As[threadIdx.x] = A[threadIdx.x]; __syncthreads();
if (threadIdx.x < 6) { As[threadIdx.x] = 99.0f; }
__syncthreads();
B[threadIdx.x] = As[threadIdx.x];
```

The pre-fill barrier is emitted at offset 0 and dropped at offset 10 — pinpointing the trigger to the nonzero start offset.

<details>
<summary>Mechanism in <code>FindConflict</code> (why the same-thread model drops the barrier)</summary>

File: `src/transform/thread_storage_sync.cc`, `TileLangThreadSyncPlanner::FindConflict`.

The fill's write is checked against the copy-in's write via the WAW call `FindConflict(writes, acc, nullptr)` ([L1234](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1234), [L1241](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1241)).

Inside `FindConflict` ([L1586](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1586)):

1. `same_access_type` is set true for `(kWrite, kWrite)` ([L1740](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1740)).
2. The WAW branch then binds **both** `prev` (copy-in) and `curr` (fill) to a **single shared thread Var** — comment: *"only need to check if the same thread overwrites its own data"* ([L1735](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1735), [L1752](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1752)).
3. Under that same-thread model the byte addresses are `prev = tx*4` (copy-in `As[tx]`) vs `curr = (tx+10)*4` (fill `As[tx+10]`). The scalar-disjointness check `CanProve(tx*4 != (tx+10)*4)` ([L1811-L1812](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1811)) is **true for all tx** → `provably_disjoint = true` ([L1774](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1774), set at [L1811](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1811)) → `range_is_overlap = false` ([L1845](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1845)) → `FindConflict` returns false → no barrier.

The same-thread-var model cannot express the real hazard (thread 10 copy-writing `As[10]` vs thread 0 filling `As[0+10]`); the `+10` offset is exactly what makes the same-`tx` addresses provably distinct.

At offset 0 the shared-index fast path fires instead: `has_same_index && same_index_means_same_byte` ([L1659-L1660](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1659)) does a Z3 constraint-equivalence test, and since the copy-in constraint is all-`tx` while the fill constraint is `tx<6`, they are inequivalent → conflict returned → barrier inserted ([L1695](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1695)), matching control (a).

**Fix direction:** for a WAW between two writes issued by *different* thread partitions (copy-in over all `tx`, fill over a strided/shifted `tx`), the disjointness test must use distinct cross-thread variables (as RAW/WAR already do at [L1752](https://github.com/tile-ai/tilelang/blob/cd37ed5f/src/transform/thread_storage_sync.cc#L1752) `else` branch) rather than a single shared `tx`, so the aliasing `thread10.As[10]` vs `thread0.As[10]` is recognized and the `__syncthreads()` is inserted.
</details>

<details>
<summary><b>Controls</b> (isolate the nonzero-offset-after-copy trigger)</summary>

All three controls are CORRECT across 3 runs, isolating the trigger to a nonzero fill offset following a copy-in:

- **(a) offset 0 fill after copy-in** `T.fill(As[0:6], 99)`: region `== [99]*6`. Generated CUDA gets a barrier *before* the fill (dropped at offset 10). Pinpoints the trigger to the nonzero start offset.
- **(b) offset 0 2D clear after copy-in** `T.clear(As)` on 8×8: all-zero (cleared as intended).
- **(c) offset fill with NO preceding copy** `T.fill(As, 0); T.fill(As[10:16], 99)`: region `== [99]*6`. With no cross-thread copy-in aliasing the offset region, the fill lands — confirming the copy-in is the necessary write-write partner.
</details>

<details>
<summary><b>Provenance</b></summary>

Reproduced independently, deterministic across 3 runs, in isolation: system `python3` (`/opt/pytorch/bin/python3`), tilelang 0.1.11, L40S sm_89, `TILELANG_CACHE_DIR`/`TMPDIR` redirected, `tilelang.disable_cache()`, top-level `@T.prim_func` + `tilelang.compile(out_idx=[1])`. Generated CUDA dumped via `get_kernel_source` for the trigger and control (a). Source line numbers are pinned to release commit `cd37ed5f` (v0.1.11); permalinks curl-checked (HTTP 200).
</details>

<details>
<summary><b>Dedup</b> (distinct from #2427 / #2556 / #2612 / #2613)</summary>

- **#2427** — `T.fill` int8 negative sign-flood: a `make_int` sub-word value-representation bug, no sync/barrier involvement. Different class (VALUE, not ordering).
- **#2556** — `ThreadPartialSyncRewriter::ProcessSharedSync` returns empty `Stmt()` for `thread_count % 32 != 0` (partial-warp tail). Here `threads = 64` (`% 32 == 0`), and the dropped barrier comes from the WAW-disjointness verdict in `FindConflict` — a different function and path.
- **#2612** — `VisitStmt_(IfThenElseNode)` hoist-and-erases an *in-loop* barrier past a `threadIdx`-dependent user `if`-guard around a write→read. This repro has no loop and no user guard (the `if(tx<6)` is the fill's own lowered predicate); the miss is a same-iteration WAW at the `FindConflict(writes, acc, nullptr)` call, not the if-hoist path.
- **#2613** — warp-specialized cross-iteration WAR on an async WGMMA operand, Hopper-only (sm_90), `num_stages=0`, two `T.ws` groups. This repro is a single warp group, no loop, no WGMMA, and reproduces on sm_89.

**New root:** cross-thread WAW write-write hazard missed by the single-shared-thread-var WAW model when a nonzero fill offset makes the same-thread byte addresses provably disjoint.
</details>

**Reach:** copy-in-then-partial-fill is ordinary, valid TileLang — `T.copy` a tensor into a shared buffer, then `T.fill`/`T.clear` a sub-region to a sentinel/pad value, then consume the buffer (a common tiling/padding idiom). The compiler accepts it (no diagnostic), lowers it (emits both the copy-in and the guarded fill), and runs it. The user never writes, and is not expected to write, a manual barrier between a `T.copy` and a `T.fill` of the same buffer — placing that `__syncthreads()` is precisely `tl.ThreadSync`'s job, which it does correctly at offset 0 (control a). Per §14 this is an **in-contract silent miscompile**; the fix is to model the cross-thread WAW and insert the barrier.
